src/ucp/core/ucp_worker.c

```diff
@@ -3066,6 +3072,13 @@ void ucp_worker_keepalive_remove_ep(ucp_ep_h ep)
 
         ucp_worker_keepalive_next_ep(worker);
         ucs_assert(worker->keepalive.iter != &ucp_ep_ext_gen(ep)->ep_list);
+
+        if (worker->keepalive.iter == worker->keepalive.iter_begin) {
+            /* If the new iterator points to the iterator from which keeaplive
+             * was started, complete keepalive, because the whole round was done
```

Comment:
/* if we moved the keepalive iterator to the stop element as a result of removing the ep, need to finish the current round */

Reply:
fixed